pilot/pkg/model/config.go

```diff
@@ -822,23 +822,23 @@ func (store *istioConfigStore) Policy(instances []*ServiceInstance, destination
 	return &out
 }
 
-func (store *istioConfigStore) DestinationRule(hostname string) *Config {
+func (store *istioConfigStore) DestinationRule(hostname Hostname) *Config {
 	configs, err := store.List(DestinationRule.Type, NamespaceAll)
 	if err != nil {
 		return nil
 	}
 
 	for _, config := range configs {
 		rule := config.Spec.(*networking.DestinationRule)
-		if ResolveShortnameToFQDN(rule.Host, config.ConfigMeta) == hostname {
+		if ResolveShortnameToFQDN(rule.Host, config.ConfigMeta).Matches(hostname) {
```

Comment:
So we need to do the most specific match here.. *.com, *.foo.com, example.foo.com